### PR TITLE
[v8]Add v8 support

### PR DESCRIPTION
**Pull request by JackBoosY**
Add google chrome v8 js engine support.

Related: #2324 #2560 #372 #1784 #2110

### CODE REVIEW

File: ports/v8/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    WORKING_DIRECTORY ${SOURCE_PATH}/v8/v8
+    LOGNAME ${TARGET_TRIPLET}
+)
+
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+    set(BUILDTYPE ia32.optdebug)
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    set(BUILDTYPE x64.optdebug)
+else()
+    message(FATAL_ERROR "Unsupported target architecture: ${VCPKG_TARGET_ARCHITECTURE}")
```
Comment (Rastaban) [inline ports/v8/portfile.cmake]: Move this test earlier so if it is run on an unsupported architecture they don't need to wait for the tool to clone before they find out.

File: ports/v8/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    WORKING_DIRECTORY ${SOURCE_PATH}/v8/v8
+    LOGNAME ${TARGET_TRIPLET}-dbg
+)
+
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+    set(BUILDTYPE ia32.release)
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    set(BUILDTYPE x64.release)
+else()
+    message(FATAL_ERROR "Unsupported target architecture: ${VCPKG_TARGET_ARCHITECTURE}")
```
Comment (Rastaban) [inline ports/v8/portfile.cmake]: This looks like a duplicate of the earlier check.

Comment (Rastaban) [reply ports/v8/portfile.cmake]: I see now, this part is building the release version and above is building the debug version.  Nevermind then.

File: ports/v8/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+vcpkg_execute_required_process(
+    COMMAND ${SOURCE_PATH}/python.bat ${SOURCE_PATH}/v8/v8/tools/dev/v8gen.py ${BUILDTYPE} -vv
+    WORKING_DIRECTORY ${SOURCE_PATH}/v8/v8
+    LOGNAME ${TARGET_TRIPLET}-dbg
+)
+
+message(STATUS "Setting configuration for shared library build")
+
+# Make DLLS
+file(APPEND ${SOURCE_PATH}/v8/v8/out.gn/${BUILDTYPE}/args.gn "is_component_build = true\n" )
```
Comment (Voskrese) [inline ports/v8/portfile.cmake]: `file(APPEND .../args.gn`
this code does not work, the file is overwritten by default settings

**Comment by JackBoosY** [comment]
Close this PR via #8963.

**Pull request closed without merging**